main.py

```diff
@@ -18,38 +25,41 @@ def parse_data(data):
 
 
 def get_houses_list(data):
-    return data['web_widgets']['post_list']
+    return data["web_widgets"]["post_list"]
 
 
 def extract_each_house(house):
-    data = house['data']
+    data = house["data"]
 
     return {
-        'title': data['title'],
-        'description': data['description'],
-        'district': data['district'],
-        'token': data['token'],
+        "title": data["title"],
+        "description": f'{data["top_description_text"]}  {data["middle_description_text"]}',
+        "district": data["action"]["payload"]["web_info"]["district_persian"],
+        "hasImage": data["image_count"] > 0,
+        "token": data["token"],
     }
 
 
 def send_telegram_message(house):
-    url = 'https://api.telegram.org/bot' + BOT_TOKEN + '/sendMessage'
-    text = f"<b>{house['title']}</b>"+"\n"
-    text += f"<i>{house['district']}</i>"+"\n"
-    text += f"{house['description']}"+"\n\n"
+    url = "https://api.telegram.org/bot" + BOT_TOKEN + "/sendMessage"
+    text = f"<b>{house['title']}</b>" + "\n"
+    text += f"<i>{house['district']}</i>" + "\n"
+    text += f"{house['description']}" + "\n"
+    if house["hasImage"]:
+        text += f"<i>تصویر : </i>" + "✅" + "\n\n"
+    else:
+        text += f"<i>تصویر : </i>" + "❌" + "\n\n"
     text += f"https://divar.ir/v/a/{house['token']}"
-
-    body = {
-        'chat_id': BOT_CHATID,
-        'parse_mode': 'HTML',
-        'text': text
-    }
-
-    requests.post(url, data=body)
+    body = {"chat_id": BOT_CHATID, "parse_mode": "HTML", "text": text}
+    result = requests.post(url, data=body)
+    if not result.status_code == 200:
+        time.sleep(5)
+        send_telegram_message(house)
```

Comment:
If we have any problem is our side, like having invalid request body, we will try infinitely.
Let's add a limit for retry.

Reply:
this recursive call is meant to address 429 (too many requests) 
I Guess its better to check for not a 200 response check for 429 response

Reply:
Imagine a case that one of the ads has some illegal characters in it or the length of description is longer than Telegram's accepted length, and Telegram API refuse it with `400` status.
In this case, we will keep resending the request.
Please add a counter and a threshold to give up on sending a request and add the ad's token into tokens array to prevent further send.

Reply:
I have changed it to only 429 response

```
    if result.status_code == 429:
        time.sleep(random.randint(3, 7))
        send_telegram_message(house)
```